src/test/java/org/killbill/billing/plugin/adyen/api/mapping/UserDataMappingServiceTest.java

```diff
@@ -0,0 +1,382 @@
+/*
+ * Copyright 2014-2015 Groupon, Inc
+ * Copyright 2014-2015 The Billing Project, LLC
+ *
+ * The Billing Project licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.killbill.billing.plugin.adyen.api.mapping;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.UUID;
+
+import org.killbill.billing.account.api.Account;
+import org.killbill.billing.payment.api.PluginProperty;
+import org.killbill.billing.plugin.adyen.client.model.UserData;
+import org.testng.annotations.Test;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class UserDataMappingServiceTest {
+
+    @Test(groups = "fast")
+    public void testToCustomerIdForCustomerId() throws Exception {
+        final String customerId = "customerId";
+        final String externalKey = "externalKey";
+        final UUID accountId = UUID.randomUUID();
```

Comment:
Yes, in general I could do that. But actually I don't like this approach in this case very much. I think the tests a better readable when each has its own distinguishable name, because I am really testing different cases here.